File: go/test/protocol/test_case_receiver.go
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
package protocol

import (
	"gopkg.in/yaml.v3"
)

type testCaseReceiver struct {
	Serializer     TestCaseSerializer `yaml:"serializer"`
	TelemetryTopic *string            `yaml:"telemetry-topic"`
	TopicNamespace *string            `yaml:"topic-namespace"`
	TopicTokenMap  map[string]string  `yaml:"topic-token-map"`
	RaiseError     TestCaseError      `yaml:"raise-error"`
}

type TestCaseReceiver struct {
	testCaseReceiver
}

func (receiver *TestCaseReceiver) UnmarshalYAML(node *yaml.Node) error {
	*receiver = TestCaseReceiver{}

	receiver.Serializer = TestCaseDefaultInfo.Prologue.Receiver.GetSerializer()
	receiver.TelemetryTopic = TestCaseDefaultInfo.Prologue.Receiver.GetTelemetryTopic()
	receiver.TopicNamespace = TestCaseDefaultInfo.Prologue.Receiver.GetTopicNamespace()

	return node.Decode(&receiver.testCaseReceiver)
}

File: dotnet/test/Azure.Iot.Operations.Protocol.MetlTests/TestCaseReceiver.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

namespace Azure.Iot.Operations.Protocol.MetlTests
{
    public class TestCaseReceiver
    {
        public static TestCaseSerializer DefaultSerializer = new();
        public static string? DefaultTelemetryTopic;
        public static string? DefaultTopicNamespace;

        public TestCaseSerializer Serializer { get; set; } = DefaultSerializer;

        public string? TelemetryTopic { get; set; } = DefaultTelemetryTopic;

        public string? TopicNamespace { get; set; } = DefaultTopicNamespace;

        public Dictionary<string, string>? TopicTokenMap { get; set; }

        public TestCaseError? RaiseError { get; set; }
    }
}
